### PR TITLE
Support for stream output of completion

### PR DESCRIPTION
**Pull request by CjangCjengh**
Stream output for completion has been enabled through function overloading
(previously, the stream function only supported messages as input)

### CODE REVIEW

File: utils/model.py [hunk 1]
```diff
@@ -542,3 +587,21 @@ def parse_messages():
...
+        input_tokens = tokenizer(prompt, return_tensors="pt")
+
+        model.generation_config.__dict__ = generation_config.__dict__
+        for response in model.generate_stream(**input_tokens.to(model.device), generation_config=generation_config):
+            token_cnt += 1
+            position = len(response)
+            yield response[start:position], None
+            start = position
+        if token_cnt == generation_config.__dict__['max_new_tokens']:
+            yield "", "length"
```
Comment (kurikomoe) [inline utils/model.py:605]: interesting, the llama.cpp thinks that when stopped by `max_new_tokens` the stop reason is `stop`

https://github.com/ggerganov/llama.cpp/pull/7198

File: utils/model.py [hunk 2]
```diff
@@ -307,6 +307,19 @@ def completion_stream(self, messages: list[dict[str, str]], generation_config: G
                 is_print_speed):
             yield output, finish_reason
 
+    def completion_stream(self, prompt: str, generation_config: GenerationConfig,
```
Comment (kurikomoe) [inline utils/model.py:310]: I am not quite in favor of `overloading` (especially in dynamic lang like python), maybe we can use another name?

**Comment by kurikomoe** [comment]
Also, Could you please provide a test script (put in scripts/ folder)? I am not quite sure about how to test this.

File: utils/model.py (hunk 2 above)
Comment (CjangCjengh) [reply utils/model.py]: Yes. You are right

**Comment by CjangCjengh** [comment]
> Also, Could you please provide a test script (put in tests/ folder)? I am not quite sure about how to test this.

Overload removed and test added

**Pull request merged**